code/game/objects/machinery/vending/vending.dm

```diff
@@ -585,6 +601,7 @@
 		else
 			return
 	SSblackbox.record_feedback("tally", "vendored", 1, R.product_name)
+	addtimer(CALLBACK(src, .proc/stock_vacuum), 2.5 MINUTES, TIMER_UNIQUE | TIMER_OVERRIDE) // We clean up some time after the last item has been vended.
```

Comment:
u need to reset that timer every time a new item is sold

Reply:
i tested and the timer gets replaced in SStimer because of both the TIMER_UNIQUE and TIMER_OVERRIDE flags